analyzer-service/calculations.py

```diff
@@ -0,0 +1,169 @@
+'''dependencies'''
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+import statsmodels.api as sm
+from datetime import datetime
+
+from DBInterface import DBInterface
+
+def filterData(tData):
+  df = pd.DataFrame.from_dict(tData)
+  df = df.T
+  df.reset_index(level=0, inplace=True)
+  x = df.loc[:,'index']
+  current_date = datetime.today()
+  filterDate = (current_date - pd.Timedelta(weeks=26)) - pd.Timedelta(days=1)
+
+  for i in range(len(df.loc[:, 'index'])):
+    df.loc[i,'index'] = datetime.strptime(df.loc[i,'index'] , '%Y-%m-%d')
+
+  return df.loc[df['index'] > filterDate]
+
+
+def sixMonthReturn(df):
+  i = len(df.loc[:, 'index']) - 1
+  change = pd.to_numeric(df.iloc[i, 5]) -  pd.to_numeric(df.iloc[0, 5])
+  pctChange = change / pd.to_numeric(df.iloc[0, 5])
+  return round(pctChange, 4)
+
+
+def loadHistorical(ticker):
+  '''
+  load historical data for class attribute - ticker
+
+  @return bool set to true if functions completes
+  '''
+  client, collection = DBInterface.connect() #need try catch here
+  cursor = collection.find_one({"MetaData.2Symbol" : "{}".format(ticker)})
```

Comment:
This is going to change when we configure pre-insertion key preprocessing (hopefully this week) but I think this is fine for now

Reply:
Yeah, easy change for when we do that.